test/files/neg/t8685.scala

```diff
@@ -0,0 +1,29 @@
+
+
+@deprecated("class C is depr", since="now")
+case class C(i: Int)
+
+case class D @deprecated("ctor D is depr", since="now") (i: Int)
+
+@deprecated("class E is depr", since="now")
+case class E(i: Int)
+@deprecated("module E is depr", since="now")
+object E
+
+@deprecated("module F is depr", since="now")
+object F {
+  case class G(i: Int)
+}
+
+trait Applies {
+  def f = C(42)
+  def g = D(42)
+  def h = E(42)
+  def i = F.G(42)
```

Comment:
Actually I was drip-feeding the fix. I realized that it's difficult/impossible as a casual contributor to visualize the pattern matching space for trees -- there's a comment somewhere that enumerates the cases -- but it would be handy if one could ask, What shapes of trees can I expect here? I didn't have time to research yesterday b/c day job. But I bet a tool could automate that (rather than debug lines).